From 28cc7d83b4480602a1457e95a81438d0d6e64d7f Mon Sep 17 00:00:00 2001
From: Richard Zhu <richard.zhu@freescale.com>
Date: Tue, 30 Jun 2015 09:58:26 +0800
Subject: [PATCH] MLK-11155-2 rpmsg: imx: set the max limit of the tty tests

samples should not live forever, set the max limit of
the tty tests.

Signed-off-by: Richard Zhu <richard.zhu@freescale.com>
---
 drivers/rpmsg/imx_rpmsg_tty.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/rpmsg/imx_rpmsg_tty.c b/drivers/rpmsg/imx_rpmsg_tty.c
index ccf783e..7ad20be 100644
--- a/drivers/rpmsg/imx_rpmsg_tty.c
+++ b/drivers/rpmsg/imx_rpmsg_tty.c
@@ -51,6 +51,12 @@ static void rpmsg_tty_cb(struct rpmsg_channel *rpdev, void *data, int len,
 		       data, len,  true);
 	rx_count++;
 
+	/* samples should not live forever */
+	if (rx_count >= MSG_LIMIT) {
+		dev_info(&rpdev->dev, "goodbye!\n");
+		return;
+	}
+
 	/* flush the recv-ed none-zero data to tty node */
 	if (strlen((unsigned char *)data) == 0)
 		return;
@@ -67,12 +73,6 @@ static void rpmsg_tty_cb(struct rpmsg_channel *rpdev, void *data, int len,
 	memcpy(cbuf, data, strlen(data));
 	tty_flip_buffer_push(&cport->port);
 	spin_unlock_bh(&cport->rx_lock);
-
-	/* samples should not live forever */
-	if (rx_count >= MSG_LIMIT) {
-		dev_info(&rpdev->dev, "goodbye!\n");
-		return;
-	}
 }
 
 static struct tty_port_operations  rpmsgtty_port_ops = { };
-- 
1.8.0

